client/src/app/site/base-repository.ts

```diff
@@ -0,0 +1,127 @@
+import { OpenSlidesComponent } from '../openslides.component';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { BaseViewModel } from './base-view-model';
+import { BaseModel, ModelConstructor } from '../shared/models/base.model';
+import { CollectionStringModelMapperService } from '../core/services/collectionStringModelMapper.service';
+
+export abstract class BaseRepository<V extends BaseViewModel, M extends BaseModel> extends OpenSlidesComponent {
+    /**
+     * Stores all the viewModel in an object
+     */
+    protected viewModelStore: { [modelId: number]: V } = {};
+
+    /**
+     * Stores subjects to viewModels in a list
+     */
+    protected viewModelSubjects: { [modelId: number]: BehaviorSubject<V> } = {};
+
+    /**
+     * Observable subject for the whole list
+     */
+    protected viewModelListSubject: BehaviorSubject<V[]> = new BehaviorSubject<V[]>(null);
+
+    /**
+     *
+     * @param baseModelCtor The model constructor of which this repository is about.
+     * @param depsModelCtors A list of constructors that are used in the view model.
+     * If one of those changes, the view models will be updated.
+     */
+    public constructor(
+        protected baseModelCtor: ModelConstructor<M>,
+        protected depsModelCtors: ModelConstructor<BaseModel>[]
+    ) {
+        super();
+
+        // Populate the local viewModelStore with ViewModel Objects.
+        this.DS.getAll(baseModelCtor).forEach((model: M) => {
+            this.viewModelStore[model.id] = this.addViewModel(model);
+            this.updateViewModelObservable(model.id);
+        });
+        this.updateViewModelListObservable();
+
+        // Could be raise in error if the root injector is not known
+        this.DS.changeObservable.subscribe(model => {
+            if (model instanceof this.baseModelCtor) {
+                // Add new and updated motions to the viewModelStore
+                this.addViewModel(model as M);
```

Comment:
Does nothing. ;-)
`this.viewModelStore[model.id] = this.addViewModel(model as M);`

Reply:
Yes, this was intended..